packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1880,11 +1880,11 @@ export class Project {
   }
 
   async persist() {
-    await this.persistLockfile();
-
-    for (const workspace of this.workspacesByCwd.values()) {
-      await workspace.persistManifest();
-    }
+    const limit = pLimit(4);
+    await Promise.all([
+      this.persistLockfile(),
+      Promise.all(this.workspaces.map(workspace => limit(() => workspace.persistManifest()))),
+    ]);
```

Comment:
Perhaps a spread to make it more readable?
```suggestion
    await Promise.all([
      this.persistLockfile(),
      ...this.workspaces.map(workspace => {
        return limit(() => workspace.persistManifest());
      }),
    ]);
```

Reply:
I wanted to avoid having to iterate the array when `Promise.all` could do it.

Reply:
Do you feel strongly against it? I still think it'd be easier to read, and I'd imagine a nested Promise.all wouldn't have better perf than a spread.

Reply:
No, I've committed the suggestion.